find_ideal_patchsize_and_magnification.py

```diff
@@ -0,0 +1,260 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py:light
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.14.1
+#   kernelspec:
+#     display_name: Python [conda env:Pytorch]
+#     language: python
+#     name: conda-env-Pytorch-py
+# ---
+
+# +
+import skimage
+import tables
+import os,sys
+import glob
+import math
+import random
+import PIL
+import numpy as np
+import cv2
+import sklearn.feature_extraction.image
+import matplotlib.pyplot as plt
+from matplotlib.patches import Rectangle
+
+# %matplotlib inline
+# %matplotlib widget
+
+# +
+# Collect image and mask filenames
+image_files=glob.glob('./Tubules/PAS/*.png')
+files = [(x, x.replace("./Tubules/PAS/","./Tubules/mask/").replace(".png","_mask.png")) for x in image_files]
+
+initial_magnification = 40 # Magnification of the original image
+
+# + tags=[]
+areas = [] # Area of each OOI
+rect_max = [] # Vertical or Horizontal dimension of each OOI (whichever is larger)
+centroids = [] # Centroid of each OOI
+image_index = [] # Which image each OOI is on
+patch_label = [] # Track mask value for each OOI
+edge_patch_thresh = 256 # Ignore OOI too close to the edge of the slide image (0 to disable)
+
+# Gather metrics from mask images
+for i,(_,fname) in enumerate(files):
```

Comment:
whats this dummy variable _ for? i think files is a list of filenames, so should be directly iteratable?

Reply:
I kept the filenames for mask and image together as tuples in the files variable. This was to eliminate the possibility that the indices would be off (e.g. using two separate lists). The first index (image file) is used further down in the code when making visuals for magnification and patch size tests.

Reply:
*nice*